### ISSUE TITLE
Get class number as new field after binning in symbology settings

### ISSUE THREAD

**Issue by krvrd** [qgis/QGIS#41352]
**Feature description.**

For some purposes, it is necessary to get the class number for the values ​​of a specific field from the attribute table. Class numbers are written as an additional attribute.
This feature might be implemented as a tool or like additional functionality (e.g. button "create a field with the class number") of layer symbology settings (for categorized and graduated type).

**Comment by gioman** [comment]
In QGIS symbology you can use the expression builder to create new attributes/classes from existing ones using an arbitrary expression (i.e. CASE/ELSE). Isn't what you are looking for?

**Comment by krvrd** [comment]
No, that's not it. I would get a new field with a class number. 

For example: 

<img width="818" alt="Снимок экрана 2021-02-04 в 13 11 37" src="https://user-images.githubusercontent.com/36080079/106878029-88e4ab80-66ea-11eb-9c0c-e9860c425179.png">

I've five classes of the Jenks binning. So I would create a new field e.g. "res_mean_jenks_class" and then fill it.

1st class — (7.5 - 7.7] — value "1" for features that have "res_mean" value in the interval (7.5 - 7.7]. 
2nd class — (7.7 - 8.125] — value "2" for features that have "res_mean" value  in the interval (7.7 - 8.125]
...
5th class — (9 - 9.5] — value "5" for features that have "res_mean" value  in the interval (7.7 - 8.125]

**Comment by gioman** [comment]
> I would get a new field with a class number.

so if I get it correctly you would a like a tool to create a new attribute (and fill it) using a class identifier obtained by symbology classes, that's it?

**Comment by krvrd** [comment]
Yes, it would really helpful tool.

**Comment by gioman** [comment]
> Yes, it would really helpful tool.

yeah, but probably it would be more suited for a plugin.

**Comment by krvrd** [comment]
I thought about it. But I've no experience in plugins creation yet.
I'll try to make a new one.
Thank you
